docs/component-stability.md

```diff
@@ -66,6 +66,74 @@ Stable components MUST be compatible between minor versions unless critical secu
 component owner MUST provide a migration path and a reasonable time frame for users to upgrade. The same rules from beta
 components apply to stable when it comes to configuration changes.
 
+#### Observability requirements
+
+Stable components should emit enough internal telemetry to let users detect errors, as well as data
+loss and performance issues inside the component, and to help diagnose them if possible.
+
+The internal telemetry of a stable component should allow observing the following:
+
+1. How much data the component receives.
+
+  For receivers, this could be a metric counting requests, received bytes, scraping attempts, etc.
```

Comment:
What would this metric measure for the [hostmetrics receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver)? Would the measurement be different depending on the OS?

Reply:
I think "scraping attempts", or maybe "number of metric points" scraped could make sense for the `hostmetrics` receiver, and those would not need to differ based on the OS.

I don't think we need to be much more specific than that, as the "data received" metrics for receiver only really seem useful if you also have a comparable "data lost" metric.

Reply:
Makes sense to me, we can look into adding builtin telemetry for the scraperhelper